tests/unit/preprocessor/RandomFourierGaussPreproc_unittest.cc

```diff
@@ -0,0 +1,37 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Author: Nanubala Gnana Sai
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/preprocessor/DensePreprocessor.h>
+#include <shogun/preprocessor/RandomFourierGaussPreproc.h>
+#include <shogun/mathematics/NormalDistribution.h>
+
+using namespace shogun;
+
+TEST(RandomFourierGaussPreproc, transform)
+{
+ const int32_t seed = 100;
+ const index_t num_vectors = 5;
+ const index_t num_features = 3;
+ const index_t target_dim = 300;
+ SGMatrix<float64_t> mat(num_vectors,num_features);
+ 
+ std::mt19937_64 prng(seed);
+ 
+ NormalDistribution<float64_t> norm_dist;
+
+ for(int i =0 ; i< num_vectors*num_features; ++i)
+    mat.matrix[i] = norm_dist(prng);
+
+ auto features = std::make_shared<DenseFeatures<float64_t>>(mat);
+ auto preproc = std::make_shared<RandomFourierGaussPreproc>();
+ preproc->set_dim_feature_space(target_dim);
+ preproc->fit(features);
+ 
+ auto preprocessed = preproc->transform(features);
+ EXPECT_EQ(preprocessed->get_feature_class(), C_DENSE);
```

Comment:
I'd remove the expect_eq

Reply:
actually this is not the best unit test, as it doesnt test anything apart from execution. It would be good to check that the results make sense ... Not sure how that is best done. You could fix the seed (against what I said above), and then project just two vectors.... and then compute their dot product and assert that this is close to the gaussian kernel between the inputs..... Note that Shoguns gaussian kernel is paramerized differently than the one in the class here (actually that shouldnt be the case, we should change it in this PR as well!). You need a seed to ensure the test doesnt fail randomly

Reply:
Took me a while to wrap my head around gaussian kernel working. Made the unit test, the error plateaus around 0.0097699474604690736(was fun watching how it decreased on increasing target_dim). I'm clueless about refactoring gaussian kernel but ofcourse, I'm fired up to do it!